pandas/tools/tile.py

```diff
@@ -187,15 +190,28 @@ def qcut(x, q, labels=None, retbins=False, precision=3):
     bins = algos.quantile(x, quantiles)
     fac, bins = _bins_to_cuts(x, bins, labels=labels,
                               precision=precision, include_lowest=True,
-                              dtype=dtype)
+                              dtype=dtype, duplicates=duplicates)
 
     return _postprocess_for_cut(fac, bins, retbins, x_is_series,
                                 series_index, name)
 
 
 def _bins_to_cuts(x, bins, right=True, labels=None,
                   precision=3, include_lowest=False,
-                  dtype=None):
+                  dtype=None, duplicates='raise'):
+
+    if duplicates not in ['raise', 'drop']:
+        raise ValueError("invalid value for 'duplicates' parameter, "
+                         + "valid options are: raise, drop")
+
+    unique_bins = algos.unique(bins)
+    if len(unique_bins) < len(bins):
+        if duplicates == 'raise':
+            raise ValueError('Bin edges must be unique: %s' % repr(bins) +
+                             ' You can drop duplicate edges ' +
+                             'by setting \'duplicates\' param')
```

Comment:
Same here about the `+`'s. Further, if you use `"` for the string, you don not need to escape the `'` inside the string. 
Lastly, can you put the injection of "repr(bins)" at the end of the string (and maybe also use `.format` instead of %)